### PR TITLE
W-13851548 oauth template policy

### PR DESCRIPTION
**Pull request by glenn-rodgers-sf**
# Writer's Quality Checklist

Before merging your PR, did you:

- [ ] Run spell checker
- [ ] Run link checker to check for broken xrefs
- [ ] Check for orphan files
- [ ] Perform a local build and do a final visual check of your content, including checking for:
  - Broken images
  - Dead links
  - Correct rendering of partials if they are used in your content
  - Formatting issues, such as:
    - Misnumbered ordered lists (steps) or incorrectly nested unordered lists
    - Messed up tables
    - Proper indentation
    - Correct header levels
- [ ] Receive final review and signoff from:
  - Technical SME
  - Product Manager
  - Editor or peer reviewer
  - Reporter, if this content is in response to a reported issue (internal or external feedback)
- [ ] If applicable, verify that the software actually got released

### CODE REVIEW

File: gateway/1.5/modules/ROOT/pages/policies-pdk-template-simple-oauth2-policy.adoc [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+      address: http://backend
+      routes:
+        - config:
+            destinationPath: /anything/echo/
+  policies:
+    - policyRef:
+        name: awesome-oauth-2-validation-v1-0-impl
+      config:
+        tokenExtractor: "#[dw::core::Strings::substringAfter(attributes.headers['Author\
+          ization'], 'Bearer ')]"
```
Comment (JaatadiaMulesoft) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-template-simple-oauth2-policy.adoc]: lets avoid this new line here if possible. 
tokenExtractor: "#[dw::core::Strings::substringAfter(attributes.headers['Authorization'], 'Bearer ')]"

**Review by JaatadiaMulesoft — APPROVED** [review]

File: gateway/1.5/modules/ROOT/pages/policies-pdk-template-simple-oauth2-policy.adoc [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+
+An antique store that buys and sells items needs to track the stock of each item in its collection and who buys it. To track the inventory movement, the company uses an API. 
+
+Because the company is franchising, they need to expose the API on the internet for all the store managers to access. To do this, they must protect the API with an authentication layer. 
+
+For their identity management provider, they hired the “Ripley 2000” Authentication service.
+
+Like many other identity providers, Ripley 2000 complies with OpenId standards and provides an introspection endpoint to validate OAuth 2.0 tokens. 
+
+The RFC 7662: Token Introspection extension specifies the request and response format of the introspection endpoint and states that an authentication mechanism is required. However the RFC 7662 does not specify which authentication mechanism should be used.
```
Comment (JennyHajee) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-template-simple-oauth2-policy.adoc]:
```suggestion
The RFC 7662: Token Introspection extension specifies the request and response format of the introspection endpoint and states that an authentication mechanism is required. However, the RFC 7662 does not specify which authentication mechanism should be used.
```

File: gateway/1.5/modules/ROOT/pages/policies-pdk-template-simple-oauth2-policy.adoc [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+
+The introspection endpoint provided by “Ripley 2000” requires a custom token to be sent in an authorization header with the request. The policy reads the token from the incoming request and sends it to the introspection endpoint to validate if the token is active. If any validation error occur, the policy logs the error and rejects the incoming request.
```
Comment (JennyHajee) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-template-simple-oauth2-policy.adoc]:
```suggestion
The introspection endpoint provided by “Ripley 2000” requires a custom token to be sent in an authorization header with the request. The policy reads the token from the incoming request and sends it to the introspection endpoint to validate if the token is active. If any validation error occurs, the policy logs the error and rejects the incoming request.
```

File: gateway/1.5/modules/ROOT/pages/policies-pdk-template-simple-oauth2-policy.adoc [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+
+To reuse the policy, the introspection endpoint url, how to extract the OAuth 2.0 token, and the custom authorization values are configurable.
```
Comment (JennyHajee) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-template-simple-oauth2-policy.adoc]:
```suggestion
To reuse the policy, the introspection endpoint URL, how to extract the OAuth 2.0 token, and the custom authorization values are configurable. 
```

File: gateway/1.5/modules/ROOT/pages/policies-pdk-template-simple-oauth2-policy.adoc [hunk 5]
```diff
@@ -0,0 +1,370 @@
...
+include::_attributes.adoc[]
+endif::[]
+:imagesdir: ../assets/images
+
+Use the simple OAuth 2.0 validation policy policy as an example of how you create a policy that makes xref:policies-pdk-configure-features.adoc#perform-an-http-call[HTTP requests].
+
+
+== Before You Begin
+
+. xref:policies-pdk-prerequisites.adoc[]
```
Comment (JennyHajee) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-template-simple-oauth2-policy.adoc]: Suggest adding something like:
Before creating a policy that makes HTTP calls, ensure you read the . xref:policies-pdk-prerequisites.adoc[].

File: gateway/1.5/modules/ROOT/pages/policies-pdk-template-simple-oauth2-policy.adoc [hunk 6]
```diff
@@ -0,0 +1,370 @@
...
+
+== Policy Use Case
+
+An antique store that buys and sells items needs to track the stock of each item in its collection and who buys it. To track the inventory movement, the company uses an API. 
+
+Because the company is franchising, they need to expose the API on the internet for all the store managers to access. To do this, they must protect the API with an authentication layer. 
+
+For their identity management provider, they hired the “Ripley 2000” Authentication service.
+
+Like many other identity providers, Ripley 2000 complies with OpenId standards and provides an introspection endpoint to validate OAuth 2.0 tokens.
```
Comment (JennyHajee) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-template-simple-oauth2-policy.adoc]:
```suggestion
Like many other identity providers, Ripley 2000 complies with OpenID standards and provides an introspection endpoint to validate OAuth 2.0 tokens. 
```

File: gateway/1.5/modules/ROOT/pages/policies-pdk-template-simple-oauth2-policy.adoc (hunk 2 above)
Comment (JennyHajee) [inline gateway/1.5/modules/ROOT/pages/policies-pdk-template-simple-oauth2-policy.adoc:24]:
```suggestion
The RFC 7662: Token Introspection extension specifies the request and response format of the introspection endpoint and states that an authentication mechanism is required. However the RFC 7662 does not specify which authentication mechanism should be used.
```

**Review by JennyHajee — APPROVED** [review]

**Pull request merged**